lib/elixir/test/elixir/module/types/descr_test.exs

```diff
@@ -2343,7 +2363,7 @@ defmodule Module.Types.DescrTest do
              )
              |> union(difference(open_map(x: atom()), open_map(x: boolean())))
              |> to_quoted_string() ==
-               "%{..., a: integer(), b: atom(), c: boolean()} or %{..., x: atom() and not boolean()}"
+               "%{..., a: integer(), b: atom(), c: boolean()} or\n  (%{..., x: atom() and not boolean()} and not %{..., a: integer(), b: atom(), c: boolean()})"
```

Comment:
I noticed something when fixing the order bug: this representation will be dependent on the order of the BDD, which itself is depending on the erlang term sorting of deep nested structures, things like

```elixir
iex> %{a: %{bitmap: 4}} < %{b: %{bitmap: 1}, a: %{bitmap: 12}}
true
```

While this is fine for the purpose of our "no duplicates" invariant, I'm worried that making the `to_quoted` relying on this order would cause brittle tests across OTP versions in the future.